platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/SerializableX509Certificate.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.crypto;
+
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.security.PublicKey;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+public class SerializableX509Certificate implements SelfSerializable {
+    public static final int MAX_CERT_LENGTH = 65_536;
```

Comment:
why this number?

Reply:
According to this [link](https://groups.google.com/g/mailing.openssl.users/c/awvGfcOwfZM) there is no max upper bound on the size of encoded certs since they may encode chains of certs.  For a self-signed cert, I was still uncertain.  The EC TLS Agreement key is signed by the RSA cert.  Given that the name and other fields in the cert are variable length, I wanted to give enough room that we would be safe to deserialize in.  I am treating the encoded cert as a blob of bytes.   

Since the encoding is DER/ASN.1, we could parse the beast and not rely on a WAG (wild ass guess) for its max size. 

Googling it looks like 4K might be enough for a single PEM file, [in the average case.](https://serverfault.com/questions/793563/what-is-the-common-size-of-ssl-certificate-give-or-take)  

I'm ignorant.  I was picking a safe value.  Would love for someone to give a more reasonable tighter bound.

Reply:
can you add a comment explaining the reasoning